Duplicated Azure instances

I get the following error when specifying Standard_D2as_v5 instances. Presumably because the [Azure catalog fetching script](https://github.com/skypilot-org/skypilot/blob/d6ce1bac7a53744b427e25ee2f40e463e5b81408/sky/clouds/service_catalog/data_fetchers/fetch_azure.py) is pulling an extra instance. The (0.086,     0.0086) price should be the correct one. https://cloudprice.net/vm/Standard_D2as_v5?tier=spot



```
  File "/python/.venv/lib/python3.11/site-packages/sky/clouds/service_catalog/common.py", line 351, in get_hourly_cost_impl
    assert region is None or len(set(df[price_str])) == 1, df
                             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
AssertionError:           InstanceType AcceleratorName  AcceleratorCount  vCPUs  MemoryGiB GpuInfo  Price  SpotPrice   Region Generation
1182  Standard_D2as_v5             NaN               NaN    2.0        8.0     NaN  0.178     0.0086  westus3      V1,V2
1183  Standard_D2as_v5             NaN               NaN    2.0        8.0     NaN  0.086     0.0086  westus3      V1,V2
```




<!-- If relevant, fill in versioning info to help us troubleshoot -->
_Version & Commit info:_
* `sky -v`: skypilot, version 1.0.0.dev20240712
* `sky -c`: skypilot, commit bce117d9ab108985baefd0866810ea291de97702

--- a/sky/clouds/service_catalog/data_fetchers/fetch_azure.py
+++ b/sky/clouds/service_catalog/data_fetchers/fetch_azure.py
@@ -140,8 +140,12 @@ def get_pricing_df(region: Optional[str] = None) -> 'pd.DataFrame':
     print(f'Done fetching pricing {region}')
     df = pd.DataFrame(all_items)
     assert 'productName' in df.columns, (region, df.columns)
-    return df[(~df['productName'].str.contains(' Windows')) &
-              (df['unitPrice'] > 0)]
+    # Filter out the cloud services and windows products.
+    # Some H100 series use ' Win' instead of ' Windows', e.g.
+    # Virtual Machines NCCadsv5 Srs Win
+    return df[
+        (~df['productName'].str.contains(' Win| Cloud Services| CloudServices'))
+        & (df['unitPrice'] > 0)]
 
 
 def get_sku_df(region_set: Set[str]) -> 'pd.DataFrame':

